Panic when promoting worker after demoting from manager

```
nnecting to "192.168.201.227:4241"
INFO[0184] Waiting for TLS certificate to be issued...
INFO[0184] Downloaded new TLS credentials with role: swarm-worker.
INFO[0184] 6487487d8193dcf6 became follower at term 3
INFO[0184] newRaft 6487487d8193dcf6 [peers: [], term: 3, commit: 43, applied: 0, lastindex: 43, lastterm: 3]
WARN[0184] ignoring request to join cluster, because raft state already exists
INFO[0184] 6487487d8193dcf6 became follower at term 3
INFO[0184] newRaft 6487487d8193dcf6 [peers: [], term: 3, commit: 43, applied: 0, lastindex: 43, lastterm: 3]
INFO[0184] Listening for local connections               addr=/tmp/swarm-node2/socket proto=unix
INFO[0184] Listening for connections                     addr=[::]:4242 proto=tcp
panic: close of closed channel

goroutine 756 [running]:
panic(0x485c760, 0xc820e718b0)
	/usr/local/opt/go/libexec/src/runtime/panic.go:481 +0x3e6
github.com/docker/swarmkit/agent.(*Node).initManagerConnection(0xc82032d760, 0x680fce8, 0xc820334e80, 0xc8204e9f20, 0x0, 0x0)
	/Users/diogo/go/src/github.com/docker/swarmkit/agent/node.go:552 +0x3bd
created by github.com/docker/swarmkit/agent.(*Node).runManager
	/Users/diogo/go/src/github.com/docker/swarmkit/agent/node.go:597 +0x697
```

--- a/agent/node.go
+++ b/agent/node.go
@@ -597,7 +597,10 @@ func (n *Node) runManager(ctx context.Context, securityConfig *ca.SecurityConfig
 			go n.initManagerConnection(ctx, ready)
 
 			go func() {
-				<-ready
+				select {
+				case <-ready:
+				case <-ctx.Done():
+				}
 				if ctx.Err() == nil {
 					n.remotes.Observe(api.Peer{NodeID: n.nodeID, Addr: n.config.ListenRemoteAPI}, 5)
 				}
@@ -611,6 +614,7 @@ func (n *Node) runManager(ctx context.Context, securityConfig *ca.SecurityConfig
 			m.Stop(context.Background()) // todo: this should be sync like other components
 			<-done
 
+			ready = nil // ready event happens once, even on multiple starts
 			n.Lock()
 			n.manager = nil
 			n.Unlock()

